spec/lib/annotate/annotate_routes_spec.rb

```diff
@@ -319,96 +319,116 @@
+    context 'When trailing annotation exists' do
+      let :route_file_content do
+        <<~EOS
 
 
 
-        ActionController::Routing...
-        foo
+          ActionController::Routing...
+          foo
 
 
-        # == Route Map
-        #
-        # another good line
-        # good line
-      EOS
+          # == Route Map
+          #
+          # another good line
+          # good line
+        EOS
+      end
 
-      expected_result = <<~EOS
+      let :expected_result do
+        <<~EOS
 
 
 
-        ActionController::Routing...
-        foo
-      EOS
+          ActionController::Routing...
+          foo
+        EOS
+      end
+
+      it 'removes trailing annotation and trim trailing newlines, but leave leading newlines alone' do
+        expect(mock_file).to receive(:puts).with(expected_result).once
+        expect(AnnotateRoutes).to receive(:puts).with(MESSAGE_REMOVED).once
 
-      expect(File).to receive(:read).with(ROUTE_FILE).and_return(route_file_content)
-      expect(mock_file).to receive(:puts).with(expected_result)
-      AnnotateRoutes.remove_annotations
+        AnnotateRoutes.remove_annotations
+      end
     end
 
-    it 'should remove prepended annotation and trim leading newlines, but leave trailing newlines alone' do
-      route_file_content = <<~EOS
-        # == Route Map
-        #
-        # another good line
-        # good line
+    context 'When prepended annotation exists' do
+      let :route_file_content do
+        <<~EOS
+          # == Route Map
+          #
+          # another good line
+          # good line
 
 
 
 
-        Rails.application.routes.draw do
-          root 'root#index'
-        end
+          Rails.application.routes.draw do
+            root 'root#index'
+          end
 
 
 
-      EOS
+        EOS
+      end
 
-      expected_result = <<~EOS
-        Rails.application.routes.draw do
-          root 'root#index'
-        end
+      let :expected_result do
+        <<~EOS
+          Rails.application.routes.draw do
+            root 'root#index'
+          end
 
 
 
-      EOS
+        EOS
+      end
 
-      expect(File).to receive(:read).with(ROUTE_FILE).and_return(route_file_content)
-      expect(mock_file).to receive(:puts).with(expected_result)
-      AnnotateRoutes.remove_annotations
+      it 'removes prepended annotation and trim leading newlines, but leave trailing newlines alone' do
+        expect(mock_file).to receive(:puts).with(expected_result).once
+        expect(AnnotateRoutes).to receive(:puts).with(MESSAGE_REMOVED).once
+
+        AnnotateRoutes.remove_annotations
+      end
     end
 
-    it 'should not remove custom comments above route map' do
-      route_file_content = <<~EOS
-        # My comment
-        # == Route Map
-        #
-        # another good line
-        # good line
-        Rails.application.routes.draw do
-          root 'root#index'
-        end
-      EOS
+    context 'When custome comments are above route map' do
```

Comment:
typo `custome`?

Reply:
Thank you for your pointing out my mistake.
I fixed it.